hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/keygen/TestCustomKeyGenerator.java

```diff
@@ -240,62 +240,6 @@ public void testInvalidPartitionKeyType(TypedProperties props) {
     }
   }
 
-  @Test
-  public void testNoRecordKeyFieldPropWithKeyGeneratorClass() {
-    testNoRecordKeyFieldProp(true);
-  }
-
-  @Test
-  public void testNoRecordKeyFieldPropWithKeyGeneratorType() {
-    testNoRecordKeyFieldProp(false);
-  }
-
-  public void testNoRecordKeyFieldProp(boolean useKeyGeneratorClassName) {
-    TypedProperties propsWithoutRecordKeyFieldProps = getPropsWithoutRecordKeyFieldProps(useKeyGeneratorClassName);
-    try {
-      BuiltinKeyGenerator keyGenerator =
-          (BuiltinKeyGenerator) HoodieSparkKeyGeneratorFactory.createKeyGenerator(propsWithoutRecordKeyFieldProps);
-
-      keyGenerator.getKey(getRecord());
-      Assertions.fail("should fail when record key field is not provided!");
```

Comment:
We can not remove all the non-key validation for the code, we still need to keep tests for `UPSERT` and null keys validation.